pkg/core/resources/model/resource.go

```diff
@@ -566,6 +566,13 @@ func ComputeLabels(
 		labels[mesh_proto.PolicyRoleLabel] = string(role)
 	}
 
+	if rd.IsProxy {
+		proxy, ok := resource.(ProxyResource)
+		if ok {
+			labels[mesh_proto.ProxyTypeLabel] = strings.ToLower(string(proxy.GetProxyType()))
```

Comment:
You can place `GetProxyType` on the ResourceSpec (not on the `Resource`). We already have an interface for policies https://github.com/lobkovilya/kuma/blob/73809a37c7835c8217880f3f8f6a7e0c786834ad/pkg/core/resources/model/resource.go#L730-L733, you can do the same for Proxy

Reply:
Right we can do this, updated